server/src/main/java/org/elasticsearch/index/engine/ThreadPoolMergeScheduler.java

```diff
@@ -449,6 +462,14 @@ long estimatedMergeSize() {
             return onGoingMerge.getMerge().getStoreMergeInfo().estimatedMergeBytes();
         }
 
+        public long getEstimateMergeMemoryBytes() {
```

Comment:
nit: I'd prefer to call this `getMergeMemoryEstimateBytes()`, similar to the provider (also rename various variales).